libs/content/workspace/src/workspace.rs

```diff
@@ -83,49 +84,100 @@ pub struct SaveResult {
 }
 
 #[derive(Clone)]
-pub struct WsConfig {
-    pub data_dir: String,
+pub struct WsPersistentStore {
+    pub path: String,
+    pub data: Arc<RwLock<WsPresistentData>>,
+}
 
-    pub auto_save: Arc<AtomicBool>,
-    pub auto_sync: Arc<AtomicBool>,
-    pub zen_mode: Arc<AtomicBool>,
+#[derive(Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct WsPresistentData {
+    pub last_open_tabs: Vec<Uuid>,
+    pub auto_save: bool,
+    pub auto_sync: bool,
 }
 
-impl Default for WsConfig {
+impl Default for WsPresistentData {
     fn default() -> Self {
-        Self {
-            data_dir: "".to_string(), // todo: potentially a bad idea
-            auto_save: Arc::new(AtomicBool::new(true)),
-            auto_sync: Arc::new(AtomicBool::new(true)),
-            zen_mode: Arc::new(AtomicBool::new(false)),
-        }
+        Self { auto_save: true, auto_sync: true, last_open_tabs: Vec::default() }
     }
 }
 
-impl WsConfig {
-    pub fn new(dir: String, auto_save: bool, auto_sync: bool, zen_mode: bool) -> Self {
-        let mut s = Self { data_dir: dir, ..Default::default() };
-        s.update(auto_save, auto_sync, zen_mode);
-        s
+impl WsPersistentStore {
+    pub fn new(path: PathBuf) -> Self {
+        match fs::File::open(&path) {
+            Ok(f) => WsPersistentStore {
+                path: path.to_string_lossy().to_string(),
+                data: Arc::new(RwLock::new(serde_json::from_reader(f).unwrap_or_default())),
+            },
+            Err(_) => WsPersistentStore {
+                path: path.to_string_lossy().to_string(),
+                data: Arc::new(RwLock::new(WsPresistentData::default())),
+            },
+        }
```

Comment:
This is strange to me for a few reasons:
- you have the owned `PathBuf` you can just hang on to that and re-open the file as needed
- you could also just hang on to the file descriptor

Reply:
It's also strange to me that if we failed to open / write to this file that we'll silently keep trying. I think at-least a log message is warranted